akka-actor/src/main/scala/akka/io/TcpConnection.scala

```diff
@@ -74,11 +74,12 @@ private[io] abstract class TcpConnection(val tcp: TcpExt, val channel: SocketCha
 
       val info = ConnectionInfo(registration, handler, keepOpenOnPeerClosed, useResumeWriting)
 
+      context.setReceiveTimeout(Duration.Undefined)
+      context.become(connected(info))
+
       // if we are in push mode or already have resumed reading in pullMode while waiting for Register
       // then register OP_READ interest
       if (!pullMode || (/*pullMode && */ !readingSuspended)) resumeReading(info, None)
-      context.setReceiveTimeout(Duration.Undefined)
-      context.become(connected(info))
```

Comment:
Looks like the right thing to do, just curious to understand more...

> resumeReading can fail, and in that case will become(unregistering)

I don't understand what is handling that error. I see unregistring via stopWith, but how is that called via resumeReading?

Reply:
`resumeReading` -> `doRead` -> `handleError` (in the 'catch' of the block starting with a call to `innerRead`) -> `stopWith`.

(I didn't find this by reading the code, but by logging a stacktrace in the repeat-job at #30510 ;) )

Reply:
right in front of me, I didn't see `doRead` - thanks